numpy/core/overrides.py

```diff
@@ -66,29 +66,28 @@
     """)
 
 
-ArgSpec = collections.namedtuple('ArgSpec', 'args varargs keywords defaults')
-
-
 def verify_matching_signatures(implementation, dispatcher):
     """Verify that a dispatcher function has the right signature."""
-    implementation_spec = ArgSpec(*getargspec(implementation))
-    dispatcher_spec = ArgSpec(*getargspec(dispatcher))
-
-    if (implementation_spec.args != dispatcher_spec.args or
-            implementation_spec.varargs != dispatcher_spec.varargs or
-            implementation_spec.keywords != dispatcher_spec.keywords or
-            (bool(implementation_spec.defaults) !=
-             bool(dispatcher_spec.defaults)) or
-            (implementation_spec.defaults is not None and
-             len(implementation_spec.defaults) !=
-             len(dispatcher_spec.defaults))):
-        raise RuntimeError('implementation and dispatcher for %s have '
-                           'different function signatures' % implementation)
-
-    if implementation_spec.defaults is not None:
-        if dispatcher_spec.defaults != (None,) * len(dispatcher_spec.defaults):
-            raise RuntimeError('dispatcher functions can only use None for '
-                               'default argument values')
+    implementation_sig = inspect.signature(implementation)
+    dispatcher_sig = inspect.signature(dispatcher)
+
+    implementation_sig_with_none_defaults = implementation_sig.replace(
+        parameters=[
```

Comment:
You've missed the return annotation that was in my original comment.

Reply:
Good catch - I had thought `signature.empty` was the default but that's incorrect

Reply:
Perhaps it is if you construct a fresh signature instead of using replace.

Reply:
That may also be faster...

Reply:
Good idea :+1:

Reply:
Okay, I got rid of the `replace` and just create the new signature directly